src/components/CommentCard.jsx

```diff
@@ -1,19 +1,48 @@
-const CommentCard = ({comment}) => {
+import { useContext, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteComment } from "../utils";
 
-const timeFormatter = new Intl.DateTimeFormat('en-GB', { hour: '2-digit', minute: '2-digit' });
-const postedDate = new Date(comment.created_at).toLocaleDateString();
-const postedTime = timeFormatter.format(new Date(comment.created_at));
+const CommentCard = ({ comment }) => {
+  const { user } = useContext(UserContext);
+  const [pendingDelete, setPendingDelete] = useState(false);
+  const [classSwitch, setClassSwitch] = useState('comment-card')
+  const [optimisticDelete, setOptimisticDelete] = useState(false)
+  const [deleteText, setDeleteText] = useState('Delete')
 
-    return (
-    <div className="comment-card">
-        
-        <p>{comment.body}</p>
-        <div className="comment-footer">
-            <div className="comment-footer-detail">by {comment.author} at {postedTime} on {postedDate}</div>
-            <div className="upvotes">{comment.votes} 👍</div>
+  const timeFormatter = new Intl.DateTimeFormat("en-GB", {
+    hour: "2-digit",
+    minute: "2-digit",
+  });
+  const postedDate = new Date(comment.created_at).toLocaleDateString();
+  const postedTime = timeFormatter.format(new Date(comment.created_at));
+
+  return (
+    <div className={classSwitch} hidden={optimisticDelete}>
+      <p>{comment.body}</p>
+      <div className="comment-footer">
+        <div className="comment-footer-detail">
+          by {comment.author} at {postedTime} on {postedDate}
+        </div>
+        <div className="comment-btns">
+          <button
+            className="delete-comment"
+            hidden={user !== comment.author}
+            onClick={() => {
+              deleteComment(comment.comment_id, setOptimisticDelete);
+              setPendingDelete(true);
+              setClassSwitch('deleting')
+              setDeleteText('Deleting...')
+            }}
+            disabled={pendingDelete}
```

Comment:
I like that you've only made the delete button visible for the author of that button 👍 
Great that you're disabling the button while the API request is pending to avoid the user mashing the button.
One small thing is that if the API request isn't successful, the delete button remains disabled even after the internet returns, so you have to refresh the page to delete the button. This isn't a huge problem or mandatory to fix, just something to consider as an optional thing to improve 🔍